### PR TITLE
Alpha/mentor approval notifications

### LINKED ISSUE #79: Mentor Approval Notification
**Description**
- After a mentor submits evidence of their internship, send notification to admins

**Requirements**
- [x] Add list of approvers (admins) to database
- [x] Create a notification for each approver
- [x] Send notification to admins after mentor evidence submission

### PR DESCRIPTION
**Pull request by OUMontiel**
# What does this PR do?
- Creates and sends notification after a mentor submits their evidence

**Related Issues**
- Closes #79 

---
**Merge Checklist**
- [ ] Branch is up to date
- [ ] No code style issues

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/MentorEvidenceServlet.java [hunk 1]
```diff
@@ -72,4 +82,36 @@ private void updateMentorEvidence(String paragraph) {
...
     }
   }
+
+  /**
+   * Adds a list of approvers (currently only the admins) to the mentor in the database.
+   */
+  private void addApprovers() {
+    int userId = Utility.getUserId();
+    // Create array to store IDs of approvers.
+    int[] approvers = {1, 2, 3};
```
Comment (shaargtz) [inline src/main/java/com/google/sps/servlets/MentorEvidenceServlet.java:92]: why does it store 1, 2, and 3?

File: src/main/java/com/google/sps/servlets/MentorEvidenceServlet.java [hunk 2]
```diff
@@ -39,8 +40,17 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
 
-    // Update mentor evidence in database.
+    // Update mentor evidence and add approvers in database.
     updateMentorEvidence(paragraph);
+    addApprovers();
+
+    // Call NotificationServlet to notify approvers.
+    response.setContentType("text/plain");
+    try {
+      request.getRequestDispatcher("/notification?type=approval&modifiedElementId=" + Utility.getUserId()).include(request, response);
```
Comment (shaargtz) [inline src/main/java/com/google/sps/servlets/MentorEvidenceServlet.java:50]: this line requires wrapping

File: src/main/java/com/google/sps/servlets/MentorEvidenceServlet.java [hunk 3]
```diff
@@ -72,4 +82,36 @@ private void updateMentorEvidence(String paragraph) {
...
+   */
+  private void addApprovers() {
+    int userId = Utility.getUserId();
+    // Create array to store IDs of approvers.
+    int[] approvers = {1, 2, 3};
+
+    for (int approverId : approvers) {
+      // Set up query to insert new experience tag to user.
+      String query = "INSERT INTO MentorApproval "
+          + "VALUES (?, ?, FALSE, FALSE)";
```
Comment (shaargtz) [inline src/main/java/com/google/sps/servlets/MentorEvidenceServlet.java:97]: you could declare the query outside since the loop does not affect its value

Comment (shaargtz) [reply src/main/java/com/google/sps/servlets/MentorEvidenceServlet.java:97]: also isn't the SQL syntax INSERT INTO Table(column1, column2...) ?

File: src/main/java/com/google/sps/servlets/MentorEvidenceServlet.java (hunk 1 above)
Comment (OUMontiel) [reply src/main/java/com/google/sps/servlets/MentorEvidenceServlet.java:93]: I hardcoded us as the approvers. I added a TODO to later pull all admins (and I think we'll need an admin table).

File: src/main/java/com/google/sps/servlets/MentorEvidenceServlet.java (hunk 2 above)
Comment (OUMontiel) [reply src/main/java/com/google/sps/servlets/MentorEvidenceServlet.java:50]: Done!

File: src/main/java/com/google/sps/servlets/MentorEvidenceServlet.java (hunk 3 above)
Comment (OUMontiel) [reply src/main/java/com/google/sps/servlets/MentorEvidenceServlet.java:98]: It is! Done.

File: src/main/java/com/google/sps/classes/SqlConstants.java [hunk 4]
```diff
@@ -18,26 +18,6 @@
...
-  // Constants used for inserting a question.
-  public static final int QUESTION_INSERT_TITLE = 1;
-  public static final int QUESTION_INSERT_BODY = 2;
-  public static final int QUESTION_INSERT_ASKERID = 3;
-
-  // Constants used for inserting a follower.
-  public static final int QUESTION_FETCH_MAXID = 1;
-  public static final int FOLLOWER_INSERT_QUESTIONID = 1;
-  public static final int FOLLOWER_INSERT_ASKERID = 2;
-
```
Comment (gloriapark) [inline src/main/java/com/google/sps/classes/SqlConstants.java:40]: Can you please update this file to avoid any unrelated changes? A lot of these changes are just identical deletions/insertions. I'm guessing this is just from merging and branching, but it would be great to take a look at the "Files changed" before submitting for review so that it makes your reviewers' lives easier :)

File: src/main/java/com/google/sps/classes/SqlConstants.java [hunk 5]
```diff
@@ -49,10 +29,6 @@
   public static final int USER_INSERT_MAJOR = 5;
   public static final int USER_INSERT_ISMENTOR = 6;
 
-  // Constants used for inserting mentor experience.
-  public static final int MENTOR_EXPERIENCE_INSERT_ID = 1;
-  public static final int MENTOR_EXPERIENCE_INSERT_TAG = 2;
-
   // Constants to update mentor evidence.
   public static final int MENTOR_EVIDENCE_PARAGRAPH = 1;
   public static final int MENTOR_EVIDENCE_USERID = 2;
```
Comment (gloriapark) [inline src/main/java/com/google/sps/classes/SqlConstants.java:34]: Are these constants still being used anywhere? If not, feel free to clean this up also!

File: src/main/java/com/google/sps/servlets/MentorEvidenceServlet.java [hunk 6]
```diff
@@ -73,4 +84,37 @@ private void updateMentorEvidence(String paragraph) {
...
   }
+
+  /**
+   * Adds a list of approvers (currently only the admins) to the mentor in the database.
+   */
+  private void addApprovers() {
+    int userId = Utility.getUserId();
+    // Create array to store IDs of approvers.
+    // TODO(oumontiel): Get IDs from all admins and remove hardcoded IDs.
+    int[] approvers = {1, 2, 3};
```
Comment (gloriapark) [inline src/main/java/com/google/sps/servlets/MentorEvidenceServlet.java:95]: Ideally these values would also be stored as constants in the SqlConstants file - They could be something like OMAR_USER_ID or something like that.

File: src/main/java/com/google/sps/servlets/NotificationServlet.java [hunk 7]
```diff
@@ -73,6 +73,11 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       query =  "SELECT follower_id FROM AnswerFollower WHERE answer_id = " + modifiedElementId;
       notificationUrl = "/question.html?id=" + modifiedElementId;
       notificationMessage = "Your answer was commented.";
+    } else if (typeOfNotification.equals("approval")) {
```
Comment (gloriapark) [inline src/main/java/com/google/sps/servlets/NotificationServlet.java:76]: It might be better to name this type of notification "requestApproval" or something more specific. I suspect you would eventually want to send a notification to the mentor who is signing up that says "You have been approved!", so it would be better to avoid having a generic notification of type "approval".

File: src/main/java/com/google/sps/classes/SqlConstants.java (hunk 4 above)
Comment (OUMontiel) [reply src/main/java/com/google/sps/classes/SqlConstants.java:40]: I tried to reorder, but I think that is best left for another PR. Undid those changes!

File: src/main/java/com/google/sps/classes/SqlConstants.java (hunk 5 above)
Comment (OUMontiel) [reply src/main/java/com/google/sps/classes/SqlConstants.java:34]: Oh, wow, they are not. Deleted!

File: src/main/java/com/google/sps/servlets/MentorEvidenceServlet.java (hunk 6 above)
Comment (OUMontiel) [reply src/main/java/com/google/sps/servlets/MentorEvidenceServlet.java:95]: Done!

File: src/main/java/com/google/sps/servlets/NotificationServlet.java (hunk 7 above)
Comment (OUMontiel) [reply src/main/java/com/google/sps/servlets/NotificationServlet.java:76]: Done!
